### ISSUE TITLE
ability to run helm deployments in parallel

### ISSUE THREAD

**Issue by jack1902** [ansible-collections/kubernetes.core#484]
##### SUMMARY

It would be great to have the ability to run numerous helm-deployments in parallel via ansible using the `async` features so that multiple tasks could be kicked off in parallel and then have a separate task to check on their status later on. Currently trying to do this results in `async is not supported for this task`

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

`kubernetes.core.helm`

##### ADDITIONAL INFORMATION

Having the ability to deploy multiple completely separate helm-charts in parallel would speed up the execution of my wider ansible role which uses other things besides helm (hence why using ansible to wire it all together).

I believe I could remove `wait: true` but then that would cause a helm-release to be successful before anything is actually running and i'd like to ensure `helm ls` reports of failed releases if a release timed out etc

```yaml
# present
- name: Deploy chart1
  kubernetes.core.helm:
    wait: true
    ...
- name: Deploy chart2
  kubernetes.core.helm:
    wait: true
    ...
- name: Deploy chart3
  kubernetes.core.helm:
    wait: true
    ...
```

```yaml
# future
- name: Deploy chart1
  async: 1000
  poll: 0
  register: chart_one_job_id
  kubernetes.core.helm:
    wait: true
    ...
- name: Deploy chart2
  async: 1000
  poll: 0
  register: chart_two_job_id
  kubernetes.core.helm:
    wait: true    
    ...
- name: Deploy chart3
  async: 1000
  poll: 0
  register: chart_three_job_id
  kubernetes.core.helm:
    wait: true
    ...

- name: wait for charts
  async_status:
    jid: "{{ item }}"
  register: job_result
  until: job_result.finished
  retries: 60
  delay: 10
  loop:
    - "{{ chart_one_job_id}}"
    - "{{ chart_two_job_id}}"
    - "{{ chart_three_job_id}}"
```

**Comment by yurnov** [comment]
Triage note — re-checked against `main` @ `e475d06` (7.0.0-dev0).

Same root cause as #678, and I don't think there's anything to implement on this side.

`plugins/action/helm.py` is a symlink to the shared action plugin, which never sets
`_supports_async` — and ansible-core's `ActionBase._supports_async` defaults to `False`, so
core rejects `async:` on any task backed by it. `git grep supports_async -- plugins/` is empty
on `main`. ansible-core's own `copy` action plugin has the same constraint.

**Suggest closing** as an ansible-core limitation.

For running many helm releases concurrently today, the practical options are spreading the
releases across hosts and raising `forks`, or `loop` + `throttle` — both of which get real
parallelism without needing `async` on a single task.

---
_Part of a sweep over all 91 open issues; index in #1215. Happy to see this reopened if it can still be reproduced on a current release._

**Comment by jack1902** [comment]
Well here is an example which seems to work around the original issue, which stems from ansible' own limitations, honestly reads horribly compared to the original layed out example if async worked but what can you do.

```yaml
---
- name: Stage parallel Helm deployments
  hosts: localhost
  gather_facts: false
  vars:
    helm_deployments:
      - name: "chart1"
        chart_ref: "ingress-nginx/ingress-nginx"
        release_namespace: "ingress"
      - name: "chart2"
        chart_ref: "bitnami/redis"
        release_namespace: "database"
      - name: "chart3"
        chart_ref: "bitnami/postgresql"
        release_namespace: "database"

  tasks:
    - name: Create dynamic local worker target for each chart
      ansible.builtin.add_host:
        name: "helm_worker_{{ item.name }}"
        groups: helm_workers
        chart_config: "{{ item }}"
        ansible_connection: local
      loop: "{{ helm_deployments }}"

- name: Deploy Helm charts concurrently using helm module
  hosts: helm_workers
  gather_facts: false
  strategy: free
  tasks:
    - name: Deploy Helm Chart
      kubernetes.core.helm:
        name: "{{ chart_config.name }}"
        chart_ref: "{{ chart_config.chart_ref }}"
        release_namespace: "{{ chart_config.release_namespace }}"
        wait: true
        state: present
```
